opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1433,21 +1467,22 @@ namespace Opm {
 
 
     template<typename TypeTag>
-    void
+    bool
     BlackoilWellModel<TypeTag>::
     updateWellControls(DeferredLogger& deferred_logger, const bool checkGroupControls)
     {
         // Even if there are no wells active locally, we cannot
         // return as the DeferredLogger uses global communication.
         // For no well active globally we simply return.
-        if( !wellsActive() ) return ;
+        if( !wellsActive() ) return false;
 
         const int episodeIdx = ebosSimulator_.episodeIndex();
         const int iterationIdx = ebosSimulator_.model().newtonMethod().numIterations();
         const auto& comm = ebosSimulator_.vanguard().grid().comm();
         updateAndCommunicateGroupData(episodeIdx, iterationIdx);
 
-        updateNetworkPressures(episodeIdx);
+        const bool local_network_changed = updateNetworkPressures(episodeIdx);
```

Comment:
We need to think about whether we should call this `updateNetworkPressures` function when `iterationIdx >= nupcol` (when the first item in NETBALAN < 0) or `iterationIdx > 0` (when NETBALAN is defaulted ).